test/api/db-handler.test.js

```diff
@@ -0,0 +1,66 @@
+jest.enableAutomock();
+jest.unmock('../../src/api/db-handler');
+
+const pg = require('pg-promise');
+
+const mockedPg = jest.fn();
+pg.mockReturnValue(mockedPg);
+const DbHandler = require('../../src/api/db-handler');
+
+const dbObject = {
+  none: jest.fn(),
+  result: jest.fn(),
+  oneOrNone: jest.fn(),
+};
+
+describe('The DB-Handler', () => {
+  beforeAll(() => {
+    mockedPg.mockReturnValue(dbObject);
+    dbObject.none.mockReturnValue(Promise.resolve(true));
+  });
+
+  it('should create a database when initialized', () => {
+    const handler = new DbHandler({
+      username: 'user',
+      password: 'secret',
+      host: '[::1]',
+      port: '1337',
+      database: 'someDb',
+    });
+
+    handler.setupDb();
+
+    expect(mockedPg).toMatchSnapshot();
+    expect(dbObject.none).toMatchSnapshot();
+  });
+
+  it('should log an error if something goes wrong', async () => {
+    global.console.error = jest.fn();
+    dbObject.none = jest.fn();
+    const testError = new Error('ERROR MESSAGE');
+    dbObject.none.mockReturnValue(Promise.reject(testError));
+    await new DbHandler({}).setupDb();
+
+    expect(global.console.error).toHaveBeenCalledTimes(2);
+    expect(global.console.error).toHaveBeenLastCalledWith(testError);
+    global.console.error.mockRestore();
+  });
+
+  it('should execute the correct call to the database on addUser', () => {
+    const handler = new DbHandler({});
+    handler.addUser('Test Email');
+    expect(dbObject.none).toMatchSnapshot();
```

Comment:
This is kind of why I thought snapshot testing was only useful for a UI.  All this test is saying is "whatever happened the first time I ran this test, it should keep happening."  There's really no assertions here that document what the expected behavior is, so reading the tests don't tell the reader what the code is actually supposed to do.

Reply:
In this case that's indeed a thing to consider, the snapshot could be replaced with 2 simple expectations.
The API class however isn't really tied to specific calls, so I used snapshots there to, as you said, reflect unintentional changes in the router.